### ISSUE TITLE
[BUG] NEXTAUTH_URL is missing/not set in Docker image

### ISSUE THREAD

**Issue by ag-gaphp** [formbricks/formbricks#2880]
### Issue Summary

The Azure AD OAuth mechanism sends what looks like a hardcoded localhost address as the redirect URI when trying to sign in. However, I am not hosted at localhost and need it to redirect to the WEBAPP_URL that I specified in the Docker Compose yml.

### Steps to Reproduce

1. Setup Azure Tenant according to the [Formbricks Docs](https://formbricks.com/docs/self-hosting/configuration#azure-sso-o-auth)
- _Observe_: issues from #2879 
2. Input the Client ID/Secret and Tenant ID to Formbricks
3. Give Microsoft Entra ID an arbitrary redirect URL to your domain (the docs don't specify what the URI should be)
4. Disable all other authentication methods
5. Spin up the Formbricks service, then go to the URL for the setup wizard
6. Attempt to sign-in using Azure
- _Observe_: Microsoft error stating that the redirect URL does not match, gives an error that appears to indicate localhost might be hardcoded somewhere (or the `WEBAPP_URL` variable does not work)

Error Text:
```plaintext
AADSTS50011: The redirect URI 'http://localhost:3000/api/auth/callback/azure-ad' specified in the request does not match the redirect URIs configured for the application 'MY-APP-UUID'. Make sure the redirect URI sent in the request matches one added to your application in the Azure portal. 
```

### Expected behavior

The redirect URL mentioned in the error message uses my defined `WEBAPP_URL`

### Other information

I am not sure yet if localhost is hardcoded in the Azure mechanism, or if the Docker setup just isn't using this variable at all for anything. I will try to take a look through the repo and report back which it might be.

### Environment

- [ ] Formbricks Cloud (app.formbricks.com)
- [X] Self-hosted Formbricks

### Desktop (please complete the following information)

- OS: `NixOS 24.05 (Uakari)`
- Docker: `v24.0.9`
- Formbricks Image: `sha256:111d8546c3c74ec84e66faf8905bae0bfc5c618b5104998cd678411c797d9bef`

**Comment by ag-gaphp** [comment]
I'm not sure where it's falling apart, but I think it has something to do with Node and how it's pulling from the environment. I can shell into my container, and `echo $WEBAPP_URL` spits out the exact string I set in my Docker Compose. So Docker clearly has the variable set correctly, it's something that isn't getting imported properly in Node.

**Comment by ag-gaphp** [comment]
I've just noticed a couple things in my logs for the web app that may be related.

1. The output after startup finished says my URL is localhost, despite the Docker env having the correct URL
```plaintext
formbricks     | All migrations have been successfully applied.
formbricks     | npm notice
formbricks     | npm notice New minor version of npm available! 10.7.0 -> 10.8.2
formbricks     | npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.8.2
formbricks     | npm notice To update run: npm install -g npm@10.8.2
formbricks     | npm notice
formbricks     |   ▲ Next.js 15.0.0-rc.0
formbricks     |   - Local:        http://localhost:3000
formbricks     |   - Network:      http://0.0.0.0:3000
formbricks     |
formbricks     |  ✓ Starting...
formbricks     |  ✓ Ready in 159ms
```

2. There is a warning from NextAuth regarding a missing URL variable
```plaintext
formbricks     | [next-auth][warn][NEXTAUTH_URL]
formbricks     | https://next-auth.js.org/warnings#nextauth_url
```

The doc page in the logs says that there needs to be a `NEXTAUTH_URL` defined. I set this variable in my Docker env and confirmed that it is set by shelling into the container. Although the NextAuth warning above disappears, I still see the localhost URL set as local in the logs.

I tested my login again, and this time the redirect URL is set properly in the error message.

## Conclusion

It seems as though the container's CMD line, or Node itself, needs to set `NEXTAUTH_URL` to whatever the `WEBAPP_URL` to potentially fix this issue.

**Comment by ag-gaphp** [comment]
Looks like this was resolved in 271ea89c8d0b6533ca20214ffd479348d3b10b98, closing
